std/algorithm/iteration.d

```diff
@@ -4280,3 +4284,111 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Finds the mean (colloquially known as the average) of a range.
+This function has $(D O(r.length)) complexity.
+
+Params:
+    r = an input range
+
+Returns:
+    the mean of r
+*/
+auto mean(R)(R r) if (isInputRange!R && !isInfinite!R)
```

Comment:
I.e., the sig constraints should at least check that range elements are convertible to floating point (which precision, though?), otherwise `+` and `0.0` will cause a compile error.